Assets/ChunkGenerator.cs

```diff
@@ -118,8 +123,69 @@ public Chunk GenerateChunk(Vector3Int position)
 		return chunk;
 	}
 
+	public Region GetOrGenerateRegion(Vector2Int position)
+	{
+		if (this.regions.TryGetValue(position, out Region region))
+		{
+			return region;
+		}
+
+		byte[] buf = BitConverter.GetBytes(position.x).Concat(BitConverter.GetBytes(position.y)).ToArray();
+		uint valueX = XXHash.CalculateHash(buf, buf.Length, seed: Convert.ToUInt32(seed));
+		uint valueZ = XXHash.CalculateHash(buf, buf.Length, seed: Convert.ToUInt32(seed + 1));
+		Vector2 relativeVoronoiPoint = new Vector2(valueX, valueZ) / uint.MaxValue;
+		Vector2 voronoiPoint = (position + relativeVoronoiPoint);
+		float humidity = (float)SimplexNoise.Scale(NoiseGenerator.Eval(voronoiPoint.x, voronoiPoint.y), 0, 1);
+		float temperature = (float)SimplexNoise.Scale(NoiseGenerator.Eval(voronoiPoint.y, voronoiPoint.x), 0, 1);
```

Comment:
Note that I changed y and x so that I do not get the same values as for the humidity